JRuby exception "Errno::ESRCH" on updating stylesheets (sass-maven-plugin v2.1)

Using this plugin with the configuration:

     <plugin>
                <groupId>nl.geodienstencentrum.maven</groupId>
                <artifactId>sass-maven-plugin</artifactId>
                <version>2.1</version>
                <executions>
                    <execution>
                        <goals>
                            <goal>update-stylesheets</goal>
                        </goals>
                        <phase>generate-resources</phase>
                    </execution>
                </executions>
                <configuration>
                    <sassSourceDirectory>src/main/stylesheets</sassSourceDirectory>
                    <destination>${project.build.directory}/css/</destination>
                    <useCompass>true</useCompass>
                    <sassOptions>
                        <style>:compressed</style>
                    </sassOptions>
                </configuration>
            </plugin>

Produces the exception on windows:

    Sass::Plugin.update_stylesheets
            at nl.geodienstencentrum.maven.plugin.sass.AbstractSassMojo.executeSassScript(AbstractSassMojo.java:237)
                at nl.geodienstencentrum.maven.plugin.sass.compiler.UpdateStylesheetsMojo.execute(UpdateStylesheetsMojo.java:52)
                at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:133)
                at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
                ... 19 more
        Caused by: javax.script.ScriptException: org.jruby.embed.EvalFailedException: (Errno::ESRCH) C:/workspace/repository/target/css/style.css
                at org.jruby.embed.jsr223.JRubyEngine.wrapException(JRubyEngine.java:104)
                at org.jruby.embed.jsr223.JRubyEngine.eval(JRubyEngine.java:93)
                at org.jruby.embed.jsr223.JRubyEngine.eval(JRubyEngine.java:142)
                at nl.geodienstencentrum.maven.plugin.sass.AbstractSassMojo.executeSassScript(AbstractSassMojo.java:231)
                ... 22 more
        Caused by: org.jruby.embed.EvalFailedException: (Errno::ESRCH) C:/workspace/repository/target/css/style.css
                at org.jruby.embed.internal.EmbedEvalUnitImpl.run(EmbedEvalUnitImpl.java:133)
                at org.jruby.embed.jsr223.JRubyEngine.eval(JRubyEngine.java:90)
                ... 24 more
        Caused by: org.jruby.exceptions.RaiseException: (Errno::ESRCH) C:/workspace/repository/target/css/style.css
                at org.jruby.RubyFile.mtime(org/jruby/RubyFile.java:941)
                at RUBY.stylesheet_needs_update?(C:/Users/user/.m2/repository/nl/geodienstencentrum/maven/sass-maven-plugin/2.1/sass-maven-plugin-2.1.jar!/gems/sass-3.4.10/lib/sass/plugin/staleness_checker.rb:65)
                at RUBY.update_stylesheets(C:/Users/user/.m2/repository/nl/geodienstencentrum/maven/sass-maven-plugin/2.1/sass-maven-plugin-2.1.jar!/gems/sass-3.4.10/lib/sass/plugin/compiler.rb:211)
                at org.jruby.RubyArray.each(org/jruby/RubyArray.java:1613)
                at RUBY.update_stylesheets(C:/Users/user/.m2/repository/nl/geodienstencentrum/maven/sass-maven-plugin/2.1/sass-maven-plugin-2.1.jar!/gems/sass-3.4.10/lib/sass/plugin/compiler.rb:209)
                at RUBY.update_stylesheets(C:/Users/user/.m2/repository/nl/geodienstencentrum/maven/sass-maven-plugin/2.1/sass-maven-plugin-2.1.jar!/gems/sass-3.4.10/lib/sass/plugin.rb:82)
                at RUBY.(root)(<script>:30)

Switching to 2.0 solved the issue.

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -19,13 +19,13 @@ install:
   - set PATH=%PATH%;C:\bin\apache-maven-%MVN_VERSION%\bin
   - IF NOT exist "C:\bin\apache-maven-%MVN_VERSION%\bin\*.*" (echo Maven %MVN_VERSION% not installed, so install it & cinst maven -Version %MVN_VERSION%) ELSE (echo Maven %MVN_VERSION% already installed)
   - cd C:\projects\sass-maven-plugin
-  - mvn install -U -Dmaven.test.skip=true -B -V -fae -q -T2
+  - mvn install -U -Dmaven.test.skip=true -B -V -fae -e -T2
 
 build: off
 
 test_script:
   - cd C:\projects\sass-maven-plugin
-  - mvn -e -DWatchMojoTestSleeptime=60000 test -fae -B
+  - mvn -e -DWatchMojoTestSleeptime=90000 test verify -fae -B
 
 environment:
   MVN_VERSION: 3.3.1

--- a/pom.xml
+++ b/pom.xml
@@ -186,6 +186,13 @@
 			<type>gem</type>
 		</dependency>
 		<dependency>
+			<groupId>rubygems</groupId>
+			<artifactId>listen</artifactId>
+			<version>3.0.2</version>
+			<scope>provided</scope>
+			<type>gem</type>
+		</dependency>
+		<dependency>
 			<groupId>commons-io</groupId>
 			<artifactId>commons-io</artifactId>
 			<version>2.4</version>

--- a/src/main/java/nl/geodienstencentrum/maven/plugin/sass/compiler/WatchMojo.java
+++ b/src/main/java/nl/geodienstencentrum/maven/plugin/sass/compiler/WatchMojo.java
@@ -30,6 +30,9 @@ import org.apache.maven.plugins.annotations.Mojo;
  */
 @Mojo(name = "watch")
 public class WatchMojo extends AbstractSassMojo {
+
+	private static final boolean IS_WINDOWS = (System.getProperty("os.name").toLowerCase().contains("win"));
+
 	/*
 	 * (non-Javadoc)
 	 *
@@ -41,6 +44,10 @@ public class WatchMojo extends AbstractSassMojo {
 		// build sass script
 		final StringBuilder sassBuilder = new StringBuilder();
 		this.buildBasicSassScript(sassBuilder);
+		if(IS_WINDOWS) {
+			sassBuilder.append("require 'listen'\n");
+			sassBuilder.append("Sass::Plugin.options.merge!(:poll => true)\n");
+		}
 		sassBuilder.append("Sass::Plugin.watch");
 		final String sassScript = sassBuilder.toString();
 

--- a/src/test/java/nl/geodienstencentrum/maven/plugin/sass/compiler/WatchMojoTest.java
+++ b/src/test/java/nl/geodienstencentrum/maven/plugin/sass/compiler/WatchMojoTest.java
@@ -97,9 +97,10 @@ public class WatchMojoTest {
 				@Override
 				public void run() {
 					try {
+						System.out.println("Start Sass watcher thread.");
 						myMojo.execute();
 					} catch (MojoExecutionException | MojoFailureException e) {
-						org.junit.Assert.fail("Execution failed: " + e);
+						org.junit.Assert.fail("Sass watcher thread execution failed: " + e);
 						this.interrupt();
 					}
 				}
@@ -107,40 +108,43 @@ public class WatchMojoTest {
 			// wait for watcher to start up...
 			System.out.println("[TEST] Waiting " + SLEEP_TIME / 1000 + " sec.");
 			this.wait(SLEEP_TIME);
+
 			// modify a file in the project
 			System.out.println("[TEST] Modify (touch) '_colours.scss'.");
 			TestResources.touch(new File(projectCopy.getAbsolutePath() + "/src/main/sass/"),
 					"_colours.scss");
+
 			// wait for watcher to catch up...
 			System.out.println("[TEST] Waiting " + SLEEP_TIME * 2 / 1000 + " sec.");
 			this.wait(SLEEP_TIME * 2);
+
 			// modify another file in the project
 			System.out.println("[TEST] Create 'print.scss'.");
 			TestResources.cp(new File(projectCopy.getAbsolutePath() + "/src/main/sass/"),
 					"compiled.scss", "print.scss");
+
 			// wait for watcher to catch up...
 			System.out.println("[TEST] Waiting " + SLEEP_TIME * 2 / 1000 + " sec.");
 			this.wait(SLEEP_TIME * 2);
 			this.notifyAll();
-
-			// done; lets check compilation results
-			TestResources.assertFileContents(projectCopy, "expected.css",
-					"target/maven-compass-test-1.0-SNAPSHOT/css/compiled.css");
-			if (!IS_WINDOWS) {
-				// skip for now because the jruby watcher fails to see the changes on windows
-				// this would be better with org.junit.Assume.assumeThat.assumeThat
-				// but since TestResources assertions are void...
-				TestResources.assertDirectoryContents(
-						new File(projectCopy.getAbsolutePath()
-							+ "/target/maven-compass-test-1.0-SNAPSHOT/css/"),
-						"compiled.css.map", "compiled.css",
-						"print.css.map", "print.css");
-				// this may fail when line endings differ, eg. on Windows
-				// set up git to check out with native file endings
-				TestResources.assertFileContents(projectCopy,
-						"print.css",
-						"target/maven-compass-test-1.0-SNAPSHOT/css/print.css");
-			}
 		}
+
+		// done; lets check compilation results
+		TestResources.assertFileContents(projectCopy, "expected.css",
+				"target/maven-compass-test-1.0-SNAPSHOT/css/compiled.css");
+
+		// skip for now because the jruby watcher fails to see the changes on windows
+		// this would be better with org.junit.Assume.assumeThat.assumeThat
+		// but since TestResources assertions are void...
+		TestResources.assertDirectoryContents(
+				new File(projectCopy.getAbsolutePath()
+					+ "/target/maven-compass-test-1.0-SNAPSHOT/css/"),
+				"compiled.css.map", "compiled.css",
+				"print.css.map", "print.css");
+		// this may fail when line endings differ, eg. on Windows
+		// set up git to check out with native file endings
+		TestResources.assertFileContents(projectCopy,
+				"print.css",
+				"target/maven-compass-test-1.0-SNAPSHOT/css/print.css");
 	}
 }

